drivers/serial/uart_npcx.c

```diff
@@ -257,7 +213,100 @@ static void uart_npcx_isr(const struct device *dev)
 		data->user_cb(dev, data->user_data);
 	}
 }
-#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+
+/*
+ * Poll-in implementation for interrupt driven config, forward call to
+ * uart_npcx_fifo_read().
+ */
+static int uart_npcx_poll_in(const struct device *dev, unsigned char *c)
+{
+	return uart_npcx_fifo_read(dev, c, 1) ? 0 : -1;
+}
+
+/*
+ * Poll-out implementation for interrupt driven config, forward call to
+ * uart_npcx_fifo_fill().
+ */
+static void uart_npcx_poll_out(const struct device *dev, unsigned char c)
+{
+	while (!uart_npcx_fifo_fill(dev, &c, 1))
+		continue;
+}
+
+#else /* !CONFIG_UART_INTERRUPT_DRIVEN */
+
+/*
+ * Poll-in implementation for byte mode config, read byte from URBUF if
+ * available.
+ */
+static int uart_npcx_poll_in(const struct device *dev, unsigned char *c)
+{
+	struct uart_reg *const inst = HAL_INSTANCE(dev);
+
+	/* Rx single byte buffer is not full */
+	if (!IS_BIT_SET(inst->UICTRL, NPCX_UICTRL_RBF))
+		return -1;
+
+	*c = inst->URBUF;
+	return 0;
+}
+
+/*
+ * Poll-out implementation for byte mode config, write byte to UTBUF if empty.
+ */
+static void uart_npcx_poll_out(const struct device *dev, unsigned char c)
+{
+	struct uart_reg *const inst = HAL_INSTANCE(dev);
+
+	/* Wait while Tx single byte buffer is ready to send */
+	while (!IS_BIT_SET(inst->UICTRL, NPCX_UICTRL_TBE))
+		continue;
+
+	inst->UTBUF = c;
+}
+#endif /* !CONFIG_UART_INTERRUPT_DRIVEN */
+
+/* UART api functions */
+static int uart_npcx_poll_in(const struct device *dev, unsigned char *c)
+{
+	struct uart_reg *const inst = HAL_INSTANCE(dev);
+
+	/* Rx single byte buffer is not full */
+	if (!IS_BIT_SET(inst->UICTRL, NPCX_UICTRL_RBF))
+		return -1;
+
+	*c = inst->URBUF;
+	return 0;
+}
+
+static void uart_npcx_poll_out(const struct device *dev, unsigned char c)
+{
+	struct uart_reg *const inst = HAL_INSTANCE(dev);
+
+	/* Wait while Tx single byte buffer is ready to send */
+	while (!IS_BIT_SET(inst->UICTRL, NPCX_UICTRL_TBE))
+		continue;
+
+	inst->UTBUF = c;
+}
```

Comment:
Please remove the duplicate functions here, or else you would have two poll_in and two poll_out functions.

Reply:
@galak Requested that I duplicate the whole function. Frustrating that GitHub doesn't keep the history, but the original version was:
```
static void uart_npcx_poll_in(...)
{
#ifdef CONFIG_UART_INTERRUPT_DRIVEN
    // Call fifo function
#else
  // Do the same as before
#endif
}
```

Reply:
The one marked here are outside of the #ifdef. So regardless of `CONFIG_UART_INTERRUPT_DRIVEN`, there are two copies of poll_in() and poll_out(). The ones after `#endif` need to be removed.

Reply:
Oh, I didn't even notice that when I did the rebase after splitting out the change. Fixed.